### PR TITLE
Update developer guide to encourage separate PRs for each part of the stack

### PR DESCRIPTION
**Pull request by chaas**
Add a section to the developer guide on changes that encourages PR authors to limit the scope of their change to one area of the stack/team as defined by CODEOWNERS.

Goal: reduce the burden on code reviewers by encouraging smaller PRs that are broken out by the team reviewing.

Open to more specific areas where this is particularly applicable. Right now the list is just the SQL parser.

Next step: sending an announcement on slack to epd-announce.

### CODE REVIEW

File: doc/developer/guide-changes.md [hunk 1]
```diff
@@ -89,6 +89,15 @@ feature in isolation. This can be particularly problematic if the bugfix or
...
 other if they are in the same RU.
 
+You should also endeavor to limit each RU to one team's scope as defined by
+the [CODEOWNERS](/.github/CODEOWNERS) file, in particular for larger changes.
+When a single RU contains changes across multiple parts of the stack, each
+reviewer has the additional overhead of digging thru the PR to figure out which
+files require their review. Splitting up the change into multiple RUs avoids
+this.\
+Areas where this is particularly applicable:
+* Changes to the [SQL parser](/src/sql-parser)
```
Comment (benesch) [inline doc/developer/guide-changes.md:99]: Can we also say "* Changes to the SQL planner" herE?

**Review by benesch — APPROVED** [review]
This is wonderful, Cara, thank you!

File: doc/developer/guide-changes.md [hunk 2]
```diff
@@ -89,6 +89,16 @@ feature in isolation. This can be particularly problematic if the bugfix or
 feature needs to be reverted—it is much more difficult to revert just one or the
 other if they are in the same RU.
 
+You should also endeavor to limit each RU to one team's scope as defined by
+the [CODEOWNERS](/.github/CODEOWNERS) file, in particular for larger changes.
+When a single RU contains changes across multiple parts of the stack, each
+reviewer has the additional overhead of digging thru the PR to figure out which
```
Comment (uce) [inline doc/developer/guide-changes.md:95]:
```suggestion
reviewer has the additional overhead of digging through the PR to figure out which
```

**Review by uce — APPROVED** [review]
LGTM.

**Pull request merged**